components/TxForm.tsx

```diff
@@ -29,6 +52,108 @@ export const TxForm = () => {
   // TODO: Placeholder notification for shielding
   //
   const { notifyUser } = useNotifications();
+  let abi = [
+    "function transfer(address,uint256) returns (bool)",
+    "function approve(address,uint256) returns (bool)",
+  ];
+  console.log(
+    getRailgunSmartWalletContractForNetwork(NetworkName.EthereumGoerli).address
+  );
+  const { isConnected, address } = useAccount();
+  const { data: signer } = useSigner();
+  const provider = useProvider();
+  const { config } = usePrepareContractWrite({
+    address: "0xb4fbf271143f4fbf7b91a5ded31805e42b2208d6",
+    abi,
+    functionName: "approve",
+    args: [
+      getRailgunSmartWalletContractForNetwork(NetworkName.EthereumGoerli)
+        .address,
+      constants.MaxUint256,
+    ],
+  });
+  const { writeAsync: doErc20Approval } = useContractWrite(config);
+
+  const doSubmit: React.FormEventHandler = async (e) => {
+    // Formatted token amounts and recipients.
+    const erc20AmountRecipients: RailgunERC20AmountRecipient[] = [
+      {
+        tokenAddress: "0xb4fbf271143f4fbf7b91a5ded31805e42b2208d6", // goerli weth
+        amountString: "0x10", // hexadecimal amount
+        recipientAddress:
+          "0zk1qyn0qa5rgk7z2l8wyncpynmydgj7ucrrcczhl8k27q2rw5ldvv2qrrv7j6fe3z53ll5j4fjs9j5cmq7mxsaulah7ykk6jwqna3nwvxudp5w6fwyg8cgwkwwv3g4", // RAILGUN address
+      },
+    ];
+
+    // The shieldPrivateKey enables the sender to decrypt
+    // the receiver's address in the future.
+    const shieldPrivateKey = keccak256(
+      await (signer as Signer).signMessage(
+        getShieldPrivateKeySignatureMessage()
+      )
+    );
+
+    // if (!doErc20Approval) throw "not prepared";
+    // await doErc20Approval();
+
+    // Public wallet to shield from.
+    const fromWalletAddress = address as `0x{string}`;
```

Comment:
Is this type correct? 
<img width="377" alt="image" src="https://user-images.githubusercontent.com/17163988/211975736-a26804f9-0360-4403-a7d1-1e7ff238a7c3.png">

Reply:
yeah, the quickstart sdk has a specific address type that we need to conform to which is `0x{string}`. Perhaps we should use some kind of [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates).